modules/auxiliary/gather/jetty_web_inf_disclosure.rb

```diff
@@ -0,0 +1,104 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Jetty WEB-INF File Disclosure',
+        'Description' => %q{
+          Jetty suffers from a vulnerability where certain encoded URIs and ambiguous paths can access
+          protected files in the WEB-INF folder. Versions effected are:
+          9.4.37.v20210219, 9.4.38.v20210224 and 9.4.37-9.4.42, 10.0.1-10.0.5, 11.0.1-11.0.5.
+          Exploitation can obtain any file in the WEB-INF folder, but web.xml is most likely
+          to have information of value.
+        },
+        'Author' => [
+          'h00die', # msf module
+          'Mayank Deshmukh', # EDB module
+          'cangqingzhe', # CVE-2021-34429
+          'lachlan roberts <lachlan@webtide.com>', # CVE-2021-34429
+          'charlesk40' # CVE-2021-28164
+        ],
+        'References' => [
+          [ 'EDB', '50438' ],
+          [ 'EDB', '50478' ],
+          [ 'URL', 'https://github.com/ColdFusionX/CVE-2021-34429' ],
+          [ 'URL', 'https://github.com/eclipse/jetty.project/security/advisories/GHSA-vjv5-gp2w-65vm' ], # CVE-2021-34429
+          [ 'URL', 'https://github.com/eclipse/jetty.project/security/advisories/GHSA-v7ff-8wcx-gmc5' ], # CVE-2021-28164
+          [ 'CVE', '2021-34429' ],
+          [ 'CVE', '2021-28164' ]
+        ],
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ ],
+          'SideEffects' => [ IOC_IN_LOGS ]
+        },
+        'DisclosureDate' => '2021-07-15',
+        'Actions' => [
+          [ 'READ_FILE', { 'Description' => 'Read file on the remote server from WEB-INF folder' } ],
+        ],
+        'DefaultAction' => 'READ_FILE'
+      )
+    )
+    register_options([
+      Opt::RPORT(80),
+      OptString.new('FILE', [false, 'File in WEB-INF to retrieve', 'web.xml']),
+      OptEnum.new('CVE', [true, 'The vulnerability to use', 'CVE-2021-34429', ['CVE-2021-34429', 'CVE-2021-28164']])
+    ])
+  end
+
+  def check
+    res = send_request_cgi('uri' => '/')
+    return Exploit::CheckCode::Safe("#{peer} - Could not connect to web service - no response") if res.nil?
```

Comment:
If the connection can't be made, that's typically a case where returning CheckCode::Unknown [would be preferred](https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method).

Reply:
I actually debated between Safe and Unknown.  Unknown's descriptions mentions a timeout, which made me think we started a connection and then later failed. like get request got a 404 or something.  NBD, I'll get it changed, I knew I'd pick the wrong one!